### PR TITLE
Add rh docker dev

### PR DESCRIPTION
**Pull request by hbrace**
# Motivation and Context

We want to bring the RH services into docker dev.
# What has changed

rh-services.yml and rh-dev.yml created to hold the RH docker services and dependant containers.
Makefile has been updated so that there are targes for `up`, `down`, `pull` and `logs` for RH and RM individually, as well as for both at the same time.
# How to test?

Test the make targets and ensure everything comes up and down as required.
Note, if bring up both RM and RH services, and then stop one or the other, the pubsub emulator will not stay up so you will need to run the `*-up` target for the services you wanted to stay up.

Run the ATs (will not work with the RH stuff up)
Run the RH cucumber tests ([with changes on the linked branch of the same name](https://github.com/ONSdigital/sdc-int-rh-cucumber/pull/28))
# Links

https://trello.com/c/zPH8WtWD/3401-bring-rh-into-our-docker-compose-setup8
# Screenshots (if appropriate):

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,51 @@
 DOT := $(shell command -v dot 2> /dev/null)
 
-up:
```
Comment (AdamHawtin) [inline Makefile:3]: Can we preserve `make up`?

File: Makefile [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
+	pipenv install --dev
+	./setup_pubsub.sh
+
+rmrh-up:
+	docker network inspect ssdcrmdockerdev_default >/dev/null || docker network create ssdcrmdockerdev_default
+	docker compose -f dev.yml -f rm-services.yml -f rh-dev.yml -f rh-services.yml up -d ${SERVICE} ;
+	pipenv install --dev
+	./setup_pubsub.sh
 	
-down:
```
Comment (AdamHawtin) [inline Makefile:9]: Can we preserve `make down` as a command to stop everything? There's no harm passing docker compose down extra docker compose files, even if they haven't been started.

File: Makefile [hunk 3]
```diff
@@ -1,19 +1,51 @@
...
+rm-down:
+	@echo "IF RH IS RUNNING, THIS WILL STOP PUBUSB EMULATOR FOR THOSE SERVICES TOO"
```
Comment (AdamHawtin) [inline Makefile:22]: Can we solve this by keeping the common services in a separate YAML and not stopping them on these specific `down` commands?

File: Makefile [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
 	docker compose -f dev.yml -f rm-services.yml down
 
-pull:
```
Comment (AdamHawtin) [inline Makefile:12]: Can we keep `make pull` as a command to pull everything?

File: Makefile [hunk 5]
```diff
@@ -1,19 +1,51 @@
...
 	docker compose -f dev.yml -f rm-services.yml pull ${SERVICE}
 
-logs:
+rh-pull:
+	docker compose -f rh-dev.yml -f rh-services.yml pull ${SERVICE}
+
+rmrh-pull:
+	docker compose -f dev.yml -f rm-services.yml -f rh-dev.yml -f rh-services.yml pull ${SERVICE}
+
+rm-logs:
```
Comment (AdamHawtin) [inline Makefile:41]: I don't think we need separate, specific `logs` targets, we could just have one which looks across all our docker compose files

File: rh-dev.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+version: "3.4"
+services:
+  pubsub-emulator:
+    container_name: pubsub-emulator
+    image: europe-west2-docker.pkg.dev/ssdc-rm-ci/docker/gcloud-pubsub-emulator:latest
+    ports:
+      - "8538:8538"
```
Comment (AdamHawtin) [inline rh-dev.yml:7]: I don't think we should duplicate the pubsub-emulator container config, better to split out the common dependencies on their own.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
All seems to be working, but I've added some suggestions on keeping the old make targets and handling the common pubsub dependency.

File: Makefile [hunk 7]
```diff
@@ -1,19 +1,51 @@
...
+	@echo "IF RM IS RUNNING, THIS WILL STOP PUBUSB EMULATOR FOR THOSE SERVICES TOO"
+	docker compose -f rh-dev.yml -f rh-services.yml down
+
+rmrh-down:
+	docker compose -f dev.yml -f rm-services.yml -f rh-dev.yml -f rh-services.yml down
+
+rm-pull:
 	docker compose -f dev.yml -f rm-services.yml pull ${SERVICE}
 
-logs:
```
Comment (AdamHawtin) [inline Makefile:15]: Could we keep a `logs` target to see everything?

**Review by AdamHawtin — APPROVED** [review]
👍 LGTM, passes RH AT's and RM's ATs (when RH isn't running).

**Review by LukeLoze1 — APPROVED** [review]

**Pull request merged**